optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -111,8 +124,118 @@ def _from_pretrained(
+                        model_id,
+                        subfolder=subfolder,
+                        revision=revision,
+                        cache_dir=cache_dir,
+                        token=token,
+                        local_files_only=local_files_only,
+                        force_download=force_download,
+                        trust_remote_code=trust_remote_code,
+                        config=config,
+                        **kwargs,
+                    )
+                    logger.info("Saved low bit model loading successfully. Other input args " "will be ignored.")
+                    return model
+                except Exception as e:
+                    logger.error(e)
+                    logger.error("Saved low bit model loading failed, please check your model.")
+                    exit(0)
+        if isinstance(quantization_config, (RtnConfig, GPTQConfig)):
+            logger.info("Applying Weight Only Quantization.")
+            warnings.warn(
+                "Weight only quantization provided by intel_extension_for_transformers is deprecated and it is provided by INC now.",
+                DeprecationWarning,
+            )
+            if is_neural_compressor_version("<=", "3.0"):
+                raise AssertionError("Please use neural_compressor version > 3.0.")
+            if is_ipex_version("<", "2.3.1") and use_xpu:
+                raise AssertionError("Please use intel_extension_for_pytorch version >= 2.3.1.")
+
+            if use_xpu:
+                # TODO: if low_cpu_mem_uasge is True, gptj will have accuracy issue on CPU device.
+                kwargs["low_cpu_mem_usage"] = True
+                kwargs["device_map"] = "cpu"
+                try:
+                    model = cls.auto_model_class.from_pretrained(
+                        model_id,
+                        *model_args,
+                        config=config,
+                        **kwargs,
+                    )
+                    model.config.update({"low_cpu_mem_usage": True})
+                except NotImplementedError:
+                    logger.info(
+                        "Failed to load models with `low_cpu_mem_usage` specified, "
+                        "will fall to traditional load method with higher memory consumption."
+                    )
+                    kwargs["low_cpu_mem_usage"] = False
+                    model = cls.auto_model_class.from_pretrained(
+                        model_id,
+                        *model_args,
+                        config=config,
+                        **kwargs,
+                    )
+                    model.config.update({"low_cpu_mem_usage": False})
+                    quantization_config.post_init_xpu()
+            else:
+                kwargs["low_cpu_mem_usage"] = True
+                model = cls.auto_model_class.from_pretrained(
+                    model_id,
+                    *model_args,
```

Comment:
same comment than in https://github.com/huggingface/optimum-intel/pull/880#discussion_r1734881040